bindings/python/CMakeLists.txt

```diff
@@ -54,16 +54,6 @@ if (CMAKE_VERSION VERSION_LESS 3.12)
     "${PROJECT_BINARY_DIR}/src")
 endif()
 
-################ OPENCV ##################
-find_package( OpenCV 4 )
-if (OpenCV_FOUND)
-  message("\nCOMPILING WITH OPENCV\n")
-  set(CMAKE_SWIG_FLAGS "-DUSE_OPENCV=1")
-  add_definitions( -DUSE_OPENCV=1 )
-else()
-  message("\nOPENCV NOT FOUND, SOME FUNCTIONALITIES WILL BE DISABLED\n")
-endif()
-
```

Comment:
This is all made unnecessary by the use of `target_link_libraries(openshot PUBLIC <opencv_related_targets>)` in the `src/CMakeLists.txt` file, which makes the target configs propagate to all library dependents automatically.

Reply:
Hi @ferdnyc I was checking your pull request and on my machine I had to include opencv on this cmake, otherwise the opencv effects are not displayed in openshot-qt

Reply:
Ok, I'll have to look at that. If that's the case, even with protocol buffers using targets, then most likely what's missing is either the compiler flag that `#define`s `USE_OPENCV` in the SWIG context, _or_ the include directories didn't get propagated to SWIG like they should've. Either way, it's definitely solvable, and will need to be solved in the target context. If any subdirectory config hacks like this are necessary, then the target config isn't right across the build which will bite us when we start generating `EXPORTED` targets from the build. (Which I'm very close to, I've already got it working in libopenshot-audio.)

I'll take a look at it in the morning. What version of CMake are you building with? (I might've asked that in the past, but I forget the answer sorry.) Some of these things do require certain versions, possibly 3.12 or 3.13 at a minimum for it to all work like it's supposed to. (But I can put in checks that use the older hacks to fake it, when they have to.)

Reply:
I'm currently using CMake 3.10.2 on my system, but I can update it if necessary

Reply:
No, that's OK, at least for the moment. The issue wasn't the version. (Though, you should update, 3.10 is ancient! Still, we theoretically support it.)

The real issue is that the `CMAKE_SWIG_FLAGS` propagation from the `src/` directory has been broken for some time, turns out. I just submitted PR #615 to sort that out. Once that's merged, the `SWIG_FLAGS` for the bindings will be set directly from the `COMPILE_DEFINITIONS` set on the `openshot` library target, so there won't be any need to play around with `CMAKE_SWIG_FLAGS` at all anymore.

Reply:
great @ferdnyc, thanks

Reply:
With CMake 3.10, you will have the issue that CMake won't automatically propagate your `INCLUDE_DIRECTORIES` from the `openshot` target. But that's OK, because you don't actually need to be setting any additional `target_include_directories` for the generated protocol buffer sources. What you should do is just always include the subdirectory when you reference them, e.g.:
```c++
#include "protobuf_messages/stabilizedata.pb.h"
```
That way, they'll be covered under the existing `INCLUDE_DIRECTORIES`. (That also means they'll work for external library consumers using the installed headers, because they'll be located at `/usr/local/include/libopenshot/protobuf_messages/`. The library's include path gets exported as `/usr/local/include/libopenshot` on install. (Or wherever it's installed.)

Reply:
In the PR branch I've fixed all the includes to use `#include "protobuf_messages/..."` and `#include "sort_filter/..."`, as well as making sure that all of the includes get installed, and that all of the libraries go the right place. (DLLs on Windows install to the `RUNTIME DESTINATION` (which is `bin/`, not `lib/`), so that needs to be set in the `install(TARGETS...)` for any shared libraries.)